app/services/session.server.ts

```diff
@@ -0,0 +1,76 @@
+import type { User } from "@prisma/client";
+import { createCookieSessionStorage, json } from "@remix-run/node";
+import invariant from "tiny-invariant";
+
+invariant(process.env.SESSION_SECRET, "SESSION_SECRET must be set");
+
+export const sessionStorage = createCookieSessionStorage({
+  cookie: {
+    name: "_session",
+    httpOnly: true,
+    path: "/",
+    sameSite: "lax",
+    secrets: [process.env.SESSION_SECRET],
+    secure: process.env.NODE_ENV === "production", // enable this in prod only
+  },
+});
+
+const USER_SESSION_KEY = "userId";
+const NONCE = "nonce";
+
+export async function getSession(request: Request) {
+  const cookie = request.headers.get("cookie");
+  return await sessionStorage.getSession(cookie);
```

Comment:
`return await` is almost always unnecessary and can actually have a minor performance hit. You can just return the promise straight up
```suggestion
  return sessionStorage.getSession(cookie);
```